tests/unit/test_idle_check.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2024 Canonical Ltd.
+# Licensed under the Apache V2, see LICENCE file for details.
+from __future__ import annotations
+
+import copy
+import json
+from typing import Any
+from unittest.mock import ANY
+
+import pytest
+
+from juju.client._definitions import (
+    FullStatus,
+)
+from juju.client.facade import _convert_response
+from juju.errors import JujuAgentError, JujuAppError, JujuMachineError, JujuUnitError
+from juju.model._idle import CheckStatus, check
+
+
+def test_check_status(full_status: FullStatus, kwargs):
+    status = check(full_status, **kwargs)
+    assert status == CheckStatus(
+        {
+            "grafana-agent-k8s/0",
+            "hexanator/0",
+            "mysql-test-app/0",
+            "mysql-test-app/1",
+        },
+        {
+            "grafana-agent-k8s/0",
+            "hexanator/0",
+            "mysql-test-app/0",
+            "mysql-test-app/1",
+        },
+        {
+            "grafana-agent-k8s/0",
+            "hexanator/0",
+            "mysql-test-app/0",
+            "mysql-test-app/1",
+        },
+    )
+
+
+def test_check_status_missing_app(full_status: FullStatus, kwargs):
+    kwargs["apps"] = ["missing", "hexanator"]
```

Comment:
Where `kwargs` is being modified before calling, what do you think about repeating yourself and just defining it in the test?
```suggestion
    kwargs = {
        apps: ["missing", "hexanator"],
        raise_on_error: False,
        raise_on_blocked: False,
        status: None,
    }
```

Reply:
Doable, but that gets rather tedious / blows up the code size.